sdk/analytics/src/main/java/com/klaviyo/analytics/Klaviyo.kt

```diff
@@ -311,4 +317,21 @@ object Klaviyo {
      */
     val Intent.isKlaviyoIntent: Boolean
         get() = this.getStringExtra("com.klaviyo._k")?.isNotEmpty() ?: false
+
+    /**
+     * Safely invoke a function and log KlaviyoExceptions rather than crash
+     */
+    private inline fun <T>safeCall(block: () -> T): T? {
+        return try {
+            block()
+        } catch (e: KlaviyoException) {
+            // KlaviyoException is self-logging
+            null
+        }
```

Comment:
We could go even farther and swallow all exceptions by adding 
```
        catch (e: Throwable) {
            Registry.log.error("Klaviyo SDK encountered an unexpected exception", e)
            null
        }
```

Reply:
>            // KlaviyoException is self-logging

I guess this statement is self explanatory but since I don't really understand android curious how the exception here is logged to console?

Reply:
In the `KlaviyoException` class, I call something like `Registry.log.error(this)` from its initializer.